File: 2021/day_2.py
class Position:
    def __init__(self):
        data_file = open("inputs/day_2_input.txt", "r")
        self.position_data = data_file.readlines()
        data_file.close()
        self.horizontal = 0
        self.depth = 0

    def plot_movement(self, movement):
        change = movement.split()
        direction = change[0]
        distance = int(change[1])

        if direction == "forward":
            self.horizontal += distance
        if direction == "backward":
            self.horizontal -= distance
        if direction == "down":
            self.depth += distance
        if direction == "up":
            self.depth -= distance

    def follow_course(self):
        for movement in self.position_data:
            self.plot_movement(movement)

        location = self.horizontal * self.depth

        return location


if __name__ == "__main__":
    course = Position()

    print(f"Location is {course.follow_course()}")

File: 2021/rust/src/day_2.rs
pub fn part_1(input: &Vec<String>) -> i32 {
    let mut depth = 0;
    let mut position = 0;

    let input_lenght = input.len();


    for i in 0..input_lenght {
        let command: Vec<&str> = input[i].split(' ').collect();
        let direction = command[0];
        let distance = command[1];

        if direction == "forward" {
            position += distance.parse::<i32>().unwrap();
        }
        if direction == "up" {
            depth -= distance.parse::<i32>().unwrap();
        }
        if direction == "down" {
            depth += distance.parse::<i32>().unwrap();
        }

    }

    let result = position * depth;
    return result;
}

pub fn part_2(input: &Vec<String>) -> i32 {
    let mut aim:i32 = 0;
    let mut depth:i32 = 0;
    let mut position:i32 = 0;

    let input_lenght = input.len();


    for i in 0..input_lenght {
        let command: Vec<&str> = input[i].split(' ').collect();
        let direction = command[0];
        let distance = command[1];

        if direction == "forward" {
            let step:i32 = distance.parse::<i32>().unwrap();
            position += step;
            depth = depth + aim * step;
        }
        if direction == "up" {
            aim -= distance.parse::<i32>().unwrap();
        }
        if direction == "down" {
            aim += distance.parse::<i32>().unwrap();
        }

    }

    let result: i32 = position * depth;
    return result;
}
